pallets/route-executor/src/lib.rs

```diff
@@ -820,11 +826,8 @@ impl<T: Config> Pallet<T> {
 		Ok(amount_in_and_outs)
 	}
 
-	fn insert_route(asset_pair: AssetPair<T::AssetId>, route: Vec<Trade<T::AssetId>>) -> DispatchResultWithPostInfo {
-		let route_as_bounded_vec: BoundedVec<Trade<T::AssetId>, sp_runtime::traits::ConstU32<MAX_NUMBER_OF_TRADES>> =
-			route.try_into().map_err(|_| Error::<T>::MaxTradesExceeded)?;
```

Comment:
What's the main issue with the previous implementation. The storage is bounded so there should be not any attack vector.

Reply:
but someone could send an encoded extrinsic with a large number of routes there. If we bound it in the extrinsic param level, we can prevent that.

Reply:
Would it be a problem? To receive such extrinsic

Reply:
in theory yes, as the data size can be very large of such encoded data. @jak-pan mentioned that there is some barrier mechanism for this within substrate which probably prevents it, we still decided to fix it